examples/wandb_sweeps/get_hyperparameters.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+"""Retrieves hyperparameters from a W&B run."""
+
+import argparse
+import logging
+
+import wandb
+import yaml
+
+import util
+
+# Top-level categories allowed in Lightning CLI configs.
+ALLOWED_TOP_LEVEL_CATEGORIES = frozenset(
+    ["data", "model", "trainer", "checkpoint", "prediction"]
+)
+
+
+def dot_to_nested_dict(flat_dict: dict[str, ...]) -> dict[str, ...]:
+    """Converts dot notation dictionary to nested dictionary.
+
+    Args:
+        flat_dict: Dictionary with dot-separated keys.
+
+    Returns:
+        Nested dictionary with proper structure.
+    """
+    nested = {}
+    for key, value in flat_dict.items():
+        if (
+            any(
+                key.startswith(f"{cat}.")
+                for cat in ALLOWED_TOP_LEVEL_CATEGORIES
+            )
+            or key in ALLOWED_TOP_LEVEL_CATEGORIES
```

Comment:
Could you just do

```
any(
                key.startswith(f"{cat}")
                for cat in ALLOWED_TOP_LEVEL_CATEGORIES
            )
```
(without the period)

Reply:
That would allow false positives like `datathingy` when I want only to match `data.thingy`.